### PR TITLE
[lldb-dap] Followup fixs for data breakpoints

### PR DESCRIPTION
**Pull request by ZequanWu**
Followup fixes to resolve comments in https://github.com/llvm/llvm-project/pull/81541

### CODE REVIEW

File: lldb/tools/lldb-dap/lldb-dap.cpp [hunk 1]
```diff
@@ -2757,13 +2769,18 @@ void request_dataBreakpointInfo(const llvm::json::Object &request) {
...
                                           : "evaluation failed");
-    } else
-      addr = llvm::utohexstr(value.GetValueAsUnsigned());
+    } else {
+      uint64_t value_as_unsigned = value.GetValueAsUnsigned();
+      if (value_as_unsigned == 0) {
+        body.try_emplace("dataId", nullptr);
+        body.try_emplace("description",
+                         "unable to evaluate expression to an address.");
+      }
```
Comment (clayborg) [inline lldb/tools/lldb-dap/lldb-dap.cpp:2778]: Avoid this and do the memory region code I show below to verify the address in somewhere we can access.

File: lldb/tools/lldb-dap/lldb-dap.cpp [hunk 2]
```diff
@@ -2757,13 +2769,18 @@ void request_dataBreakpointInfo(const llvm::json::Object &request) {
...
+      addr = llvm::utohexstr(value_as_unsigned);
+    }
   } else {
-    auto state = g_dap.target.GetProcess().GetState();
     body.try_emplace("dataId", nullptr);
-    body.try_emplace("description",
-                     "variable not found: " + llvm::utostr(state));
+    body.try_emplace("description", "variable not found");
```
Comment (clayborg) [inline lldb/tools/lldb-dap/lldb-dap.cpp:2783]: Do we want more info shown here like the variable name that we tried to lookup?

Comment (ZequanWu) [reply lldb/tools/lldb-dap/lldb-dap.cpp:2795]: Do we have a way to get the complete name? 
The `name` passed by the request is a child member name and variableReference only tells us its parent. For watching `a.b.c`, we only gets the name `c` and variableReference value for `c`'s parent which is `b`.

**Review by ZequanWu — COMMENTED** [review]
In terms of interpreting `name` as `${number of bytes}@${expression}`, it's made-up. And I just noticed that someone already filed an issue about this: https://github.com/microsoft/debug-adapter-protocol/issues/455. Should we revert the change inside the condition: `variablesReference == 0 && frame.IsValid()`?

**Pull request closed without merging**